frame/babe/src/lib.rs

```diff
@@ -443,7 +448,15 @@ impl<T: Trait> Module<T> {
 				Self::deposit_consensus(ConsensusLog::NextEpochData(next))
 			}
 
-			CurrentSlot::put(digest.slot_number());
+			// the slot number of the current block being initialized
+			let current_slot = digest.slot_number();
+
+			// how many slots were skipped between current and last block
+			let lateness = current_slot.saturating_sub(CurrentSlot::get() + 1);
+			let lateness = T::BlockNumber::from(lateness as u32);
+
+			Lateness::<T>::put(lateness);
```

Comment:
We're trying to cut down on extra DB I/O as it's pretty slow. Could we get this merged into `CurrentSlot`?

Reply:
This is always killed in `on_finalize` and should never reach the DB.